parlai/scripts/generate_model_card.py

```diff
@@ -0,0 +1,1455 @@
+    return table
+
+
+def make_html_table(rows, header):
+    table = ['<table><tr><th>' + '</th><th>'.join(header) + '</th></tr>']
+    for row in rows:
+        table.append('<tr><td>' + '</td><td>'.join(row) + '</td></tr>')
+    table.append('</table>')
+    return "\n".join(table)
+
+
+#################################
+# Graphing functions
+#################################
+
+
+def get_heatmap(stats_dfs, title=None, tfsize=16, heatmapkws_user=None, fout=None):
+    # get vmax and vmin and step
+    tmp_max = max([df.max().max() for df in stats_dfs])
+    tmp_min = min([df.min().min() for df in stats_dfs])
+    step = 5 if (tmp_max - tmp_min) < 0.5 else 10
+    vmax = step * math.ceil(tmp_max * 100 / step) / 100
+    vmin = step * math.floor(tmp_min * 100 / step) / 100
+    # create dictionary for heatmap args and update any args passed by user
+    cmap = sns.color_palette("Oranges", as_cmap=True)
+    heatmapkws = {
+        'vmin': vmin,
+        'vmax': vmax,
+        'annot': True,
+        'linecolor': 'black',
+        'cmap': cmap,
+        'linewidths': 0.75,
+        'fmt': ".2%",
+    }
+    if heatmapkws_user:
+        heatmapkws.update(heatmapkws_user)
+
+    # create subplots
+    ratios = [df.shape[0] for df in stats_dfs]
+    N = len(stats_dfs)
+    fig, axs = plt.subplots(nrows=N, gridspec_kw={'height_ratios': ratios}, sharex=True)
+
+    # add color bar (with the last subplot)
+    # left, bot, width, height
+    cbar_ax = fig.add_axes([0.99, 0.15, 0.03, 0.7])
+    for i, df in enumerate(stats_dfs):
+        ax = sns.heatmap(
+            df,
+            ax=axs[i],
+            xticklabels=i == (N - 1),
+            **heatmapkws,
+            cbar=i == (N - 1),
+            cbar_ax=None if i != (N - 1) else cbar_ax,
+        )
+        # remove bottom ticks, set the borders
+        ax.tick_params(bottom=False)
+        ax.patch.set_edgecolor('black')
+        ax.patch.set_linewidth('1')
+        # to be safe, make sure y-axis is in the right direction
+        _ = ax.set_yticklabels(ax.get_yticklabels(), rotation=0)
+        if i == (N - 1):
+            # rotate x-axis label{}s if last subplot
+            labels = ax.get_xticklabels()
+            _ = ax.set_xticklabels(labels, rotation=45, horizontalalignment='right')
+            # adjust cbar labels if last subplot
+            ticks = {i / 100: f"{i}%" for i in range(0, int(vmax * 100 + 1), step)}
+            cbar = ax.collections[0].colorbar
+            cbar.set_ticks(list(ticks.keys()))
+            cbar.set_ticklabels(list(ticks.values()))
+    if title:
+        fig.suptitle(title, fontsize=tfsize)
+    if fout:
+        fig.savefig(fout, bbox_inches="tight")
+    return fig, axs
+
+
+#################################
+# Setup-related functions
+#################################
+
+
+def setup_args(parser=None) -> ParlaiParser:
+    if parser is None:
+        parser = ParlaiParser(True, True, 'Automatically generate the model card')
```

Comment:
if parser if None:
    parser = ParlaiParser(True, True, 'Automatically generate the model card')
parser.add_argument_group(Automatically generate the model card")
parser.add_argument(....)
...
return parser

parser = eval_model.setup_args()
parser = setup_args(parser)